mapper/mapper.go

```diff
@@ -62,7 +62,7 @@ func createTableOfContents(
 		displayOrder = append(displayOrder, "publications")
 	}
 
-	if len(relatedDatasets) > 0 || len(relatedAPIDatasets) > 0 {
```

Comment:
I could not find the ticket with the actual requirements for this so just checking that we do want to display the api datasets in a different section to the "old" datasets? It seems strange to me

Does this change mean `relatedAPIDatasets` is no longer in use? It looks like we use `relatedArticleDatasets` to determine whether to show the section displaying api datasets, despite it actually using `RelatedAPIDatasets` to display the data. I think we should only have one of them